Added documentation for the throw-launch feature. (#2897)
PX4/PX4-user_guide: en/features_mc/throw_launch.md

# Throw Launch (Multicopter)

:::warning Experimental
This feature was introduced after PX4 v1.14.

- It has not yet been broadly tested on different vehicle configurations or scenarios. various platforms and in different scenarios.
- The majority of testing has been done in position mode.
  Other modes should also work.

:::

This feature allows a multicopter to be started by arming it from a fixed position and then throwing it into the air.
The vehicle then turns on the motors and operates according to its current mode.

> The vehicle detects that it has been thrown based on reaching a certain speed and altitude (5m/s and 2m by default, respectively) and then starts the motors at the apex of the throw, once it determines that it has started to fall.

Author: Actually, altitude is not checked, only the speed. I've mentioned that the drone should be thrown "about 2m away and 2m up" just as an example of a throw that results in a speed of 5m/s. Do you think it's not clear that was only an example and should be clarified?

Reviewer: @zeroos Absolutely - because I read it as a hard limit.
The point though is that we want to throw fast enough to invoke the throw trigger, and high enough that the motors can start before the vehicle falls to the height of a person.
Is 2 m likely to be high enough in most cases?

Author: It depends on a platform, but I feel like for most multicopters small enough to be thrown, it should be fine, unless they are very limited in power. I'll try to make it clear in the modification.

Reviewer: No need, I have fixed it :-)

> Note that unlike in other configurations, when throw launch is enabled the propellers should not spin until the launch is detected ( even when the vehicle is armed).

Author: Should we also mention that with throw mode enabled, it repeats the arming sound until the vehicle is thrown (or disarmed)?

> The vehicle will not arm while it is moving, and should not be used on a moving platform (this is to reduce the chance of accidental triggering if a vehicle is armed and then moved to a launch point).

Author: The first part of the sentence is probably true (I did not modify the arming conditions in any way), but in the initial comment about "moving platform" I was trying to highlight something else. I propose to change this sentence in the following way, please let me know if it's more clear for you now:

> The vehicle should not be transported after being armed and before the throw. In particular, the throw should not be executed from a moving platform. The reason behind this is that the condition to start the motors depends on absolute speed of the multicopter and does not account for any additional movement. Trying to throw the drone from a moving platform might result in the motors being started prematurely.

> As usual the vehicle will disarm if the throw is not detected within a few seconds of arming.

Reviewer: Please check this. My assumption is that your safety feature is "will not arm while moving".

Author: From what I understand, that's the default behavior (not related to throw mode). The problems might occur when the vehicle moves after arming (I've proposed a description of these problems here: https://github.com/PX4/PX4-user_guide/pull/2897#discussion_r1402955353 ). Do you think we should also mention that the vehicle does not arm while moving?

> As usual the vehicle will disarm if the throw is not detected within a few seconds of arming.

Author: That's not true, when the throw mode is enabled, automatic disarming does not trigger. It happened once to us that the drone disarmed just before the drone and then we introduced this change.

# Throw Launch (Multicopter)

:::warning Experimental
This feature was introduced after PX4 v1.14.

> - It has not yet been broadly tested on different vehicle configurations or scenarios. various platforms and in different scenarios.

Reviewer 2: ```suggestion
- It has not yet been broadly tested on different vehicle configurations or scenarios.
```

Reviewer: Whoops. Thanks.

> - The majority of testing has been done in position mode.
>   Other modes should also work.

Reviewer 2: ```suggestion
- The majority of testing has been done in position mode. Other modes should also work.
```

Reviewer: I like things on separate lines.

:::

This feature allows a multicopter to be started by arming it from a fixed position and then throwing it into the air.
The vehicle then turns on the motors and operates according to its current mode.

> The vehicle detects that it has been thrown based on reaching a certain speed (5m/s), and then starts the motors at the apex of the throw, once it determines that it has started to fall.

Reviewer 2: ```suggestion
The vehicle detects that it has been thrown based on reaching a certain speed (default 5m/s), and then starts the motors at the apex of the throw, once it determines that it has started to fall.
```

Reviewer: I put default because that speed is a default. I guess it doesn't matter since the fact it can be changed is clear lower down.

> You need to throw it high enough so that the vehicle can stabilize its height well before falling in range of people (2 or 3 metres is usually sufficient).

Author: This might suggest to some that it's okay to throw the drone 2m away from people, I try to keep at least 10-20m even when flying normally... Maybe we can just remove "in range of people"?

Reviewer 2: I agree maybe "in range of any obstacles" is less tempting to bring people close.

Reviewer: Good catch. It really depends on your vehicle - if you had a small drone with prop guards it wouldn't matter. Let's do this.

```suggestion
You need to throw it high enough so that the vehicle can stabilize its height well before falling anywhere near people.
```

Note that unlike in other configurations, when throw launch is enabled the propellers should not spin until the launch is detected (even when the vehicle is armed), and that the arming tone will continue to play until the vehicle is thrown.
The vehicle will not automatically disarm after arming, and must be manually disarmed if you choose not to throw it. 

The vehicle should not be transported after being armed and before the throw.
In particular, the throw should not be executed from a moving platform.
The reason for this is that the condition to start the motors depends on absolute speed of the multicopter and does not account for any additional movement.
Trying to throw the drone from a moving platform might result in the motors being started prematurely.

## Safety

:::warning

> Throw launch is potentially dangerous as it requires the operator to hold an armed multicopter and be in proximity when it is flying.

Reviewer 2: ```suggestion
Throw launch is dangerous as it requires the operator to hold an armed multicopter and be in proximity when it is flying.
```
Let's be honest.

Reviewer: I've played with a throwable caged rotor copter at about 200g that wasn't scary. But I take your point.

Reviewer 2: I destroyed the keyboard of my laptop using a 35g micro quad. I agree it's less dangerous but if my finger (or eye) would have taken that hit that would have lead to injury...

Reviewer: Ouch!
